### DISCUSSION TITLE
Unregister array fields

### DISCUSSION THREAD

**Discussion by RonneiPeterson** [react-hook-form/react-hook-form#8794] [Q&A]
I use array fields and i need unregister or delete especific indexes, example, i have a list of products and  i need build a new list but with minus items, unregister not work all times. Example

`    unregister('projetos.produtos'); //this dont work all times
    novoObjFull.produtos?.map((itemProduto, index) => {
      setValue(`projetos.produtos[${index}].id`, itemProduto.id);
      setValue(`projetos.produtos[${index}].formula_quantidade`, itemProduto.formula_quantidade);
`

**Comment by RonneiPeterson** [comment]
I resolved using empty array before setValue

`
    setValue(`projetos.produtos[]`,[]);

    novoObjFull.produtos?.map((itemProduto, index) => {
      setValue(`projetos.produtos[${in`

**Reply by bluebill1049** [reply]
did you unmount the entire `useFieldArray` as well?
